drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/conversion/GuidedDecisionTablePopulater.java

```diff
@@ -116,70 +118,131 @@ private void processIndirectSourceBuilders(final int maxRowCount) {
+                        allPatternsHaveTemplateKeys = false;
+                    }
+                }
+
+                // Copy definition to column.
+                // If ALL patterns in the iteration have Template keys we can use IPattern otherwise fallback to FreeFormLine
+                if (allPatternsHaveTemplateKeys) {
+                    for (int iPattern = previousPatternCount; iPattern < rm.lhs.length; iPattern++) {
+                        final IPattern pattern = rm.lhs[iPattern];
+                        column.getDefinition().add(pattern);
+                    }
+                } else {
+                    final FreeFormLine ffl = new FreeFormLine();
+                    ffl.setText(lhsBuilders.get(i).getResult());
+                    column.setDefinition(Collections.singletonList(ffl));
+                }
+
+                //Copy variables to column
+                for (int iPattern = previousPatternCount; iPattern < rm.lhs.length; iPattern++) {
+                    final IPattern pattern = rm.lhs[iPattern];
+                    final Map<InterpolationVariable, Integer> templateKeys = patternTemplateKeys.get(pattern);
+                    if (Objects.nonNull(templateKeys)) {
+                        final List<InterpolationVariable> ivs = new ArrayList<>(templateKeys.keySet());
+                        for (BRLVariableColumn variableColumn : variableColumns) {
+                            final Iterator<InterpolationVariable> ivsIts = ivs.iterator();
+                            while (ivsIts.hasNext()) {
+                                final InterpolationVariable iv = ivsIts.next();
+                                if (iv.getVarName().equals(variableColumn.getVarName())) {
+                                    final BRLConditionVariableColumn source = (BRLConditionVariableColumn) variableColumn;
+                                    final BRLConditionVariableColumn target = makeBRLConditionVariableColumn(source, iv);
+                                    column.getChildColumns().add(target);
+                                    ivsIts.remove();
+                                }
+                            }
                         }
                     }
                 }
 
+                //Give the column a header
                 if (column.getChildColumns().isEmpty()) {
                     setZeroParameterConditionColumnHeader(column,
                                                           variableColumns);
                 } else {
                     setCompositeColumnHeader(column);
                 }
             }
+            previousPatternCount = rm.lhs.length;
+        }
+
+        //-- PROCESS RHS ACTIONS
+        final StringBuilder rule = new StringBuilder();
+        if (!(dmo.getPackageName() == null || dmo.getPackageName().isEmpty())) {
+            rule.append("package ").append(dmo.getPackageName()).append("\n");
+        }
+        rule.append("rule 'temp' \n").append("when \n");
```

Comment:
We do the same for LHS patterns, isn't it worth to separate method?

Reply:
Done.